### PR TITLE
[CELEBORN-884][BUILD] Consolidate all dependencies into a global object `Dependencies`

### PR DESCRIPTION
**Pull request by cfmcgrady**
### What changes were proposed in this pull request?

As title

### Why are the changes needed?

consolidate all sbt dependencies into a global object `Dependencies`, similar to Maven's dependencyManagement, to improve dependency management.

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

Pass GA

### CODE REVIEW

File: project/CelebornBuild.scala [hunk 1]
```diff
@@ -55,20 +43,81 @@ object CelebornCommonSettings {
   val guavaVersion = "14.0.1"
   val hadoopVersion = "3.2.4"
   val javaxServletVersion = "3.1.0"
+  val junitInterfaceVersion = "0.13.3"
+  // don't forget update `junitInterfaceVersion` when we upgrade junit
```
Comment (pan3793) [inline project/CelebornBuild.scala:47]: seems no strong contract?

File: project/CelebornBuild.scala [hunk 2]
```diff
@@ -334,14 +383,14 @@ object CelebornWorker {
...
-        "io.netty" % "netty-all" % nettyVersion,
-        "org.apache.logging.log4j" % "log4j-1.2-api" % log4j2Version,
-        "org.apache.logging.log4j" % "log4j-slf4j-impl" % log4j2Version,
-        "org.fusesource.leveldbjni" % "leveldbjni-all" % leveldbJniVersion,
-        "org.roaringbitmap" % "RoaringBitmap" % roaringBitmapVersion,
-        "org.mockito" %% "mockito-scala-scalatest" % scalatestMockitoVersion % "test",
+        Dependencies.guava,
+        Dependencies.commonsIo,
+        Dependencies.ioNetty,
+        Dependencies.log4j12Api,
```
Comment (pan3793) [inline project/CelebornBuild.scala:389]: it's not clear to me the scope of each dep, and there is possible that a dep may have different scopes in different modules

File: project/CelebornBuild.scala (hunk 1 above)
Comment (cfmcgrady) [reply project/CelebornBuild.scala:47]: 1. The JUnit interface delineates its JUnit version compatibility matrix within its [documentation](https://github.com/sbt/junit-interface/#setup). Therefore, I think it's best to keep it consistent with the documentation.

| junit-interface version | JUnit version |
|------------------------|---------------|
| 0.13.2                 | 4.13.2        |
| 0.13.1                 | 4.13.1        |
| 0.13                   | 4.13          |
| 0.12                   | 4.12          |

2. As per the [SBT documentation](https://www.scala-sbt.org/1.x/docs/Testing.html#JUnit), if we're planning to upgrade from JUnit 4 to JUnit 5, we'll need to switch from using the sbt JUnit interface to the sbt-jupiter-interface.

File: project/CelebornBuild.scala (hunk 2 above)
Comment (cfmcgrady) [reply project/CelebornBuild.scala:389]: In the current implementation, we're using a naming pattern like this:

```
${dependency}${scope}
```

For instance:

```
`log4j12Api` -> "org.apache.logging.log4j" % "log4j-1.2-api" % log4j2Version 
`log4j12ApiTest` -> "org.apache.logging.log4j" % "log4j-1.2-api" % log4j2Version % "test"
```
Moreover, a dependency lacking a scope suffix designates the default runtime scope.

Do you have a better suggestion for the name rules?

Comment (pan3793) [reply project/CelebornBuild.scala:389]: I like the latter one

File: project/CelebornBuild.scala (hunk 1 above)
Comment (pan3793) [reply project/CelebornBuild.scala:47]: I see, it makes sense, thanks for explaination

File: project/CelebornBuild.scala (hunk 2 above)
Comment (cfmcgrady) [reply project/CelebornBuild.scala:387]: updated

**Review by pan3793 — APPROVED** [review]

**Comment by cfmcgrady** [comment]
thanks, merging to main(v0.4.0).

**Pull request closed without merging**